### PR TITLE
chore: [Multi-origin] Correctly find the spec bridge if chromeWebSecurity set to false.

### PR DESCRIPTION
**Pull request by mjhenkes**
- Closes 

### User facing changelog

n/a

### Additional details

If chromeWebSecurity is set to false we can't rely on a security error to identify the correct spec bridge.

This PR adds some check to verify we are on the correct origin policy for the spec bridge.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/runner/injection/multi-domain.js [hunk 1]
```diff
@@ -16,7 +16,18 @@ const findCypress = () => {
     try {
       // If Cypress is defined and we haven't gotten a cross origin error we have found the correct bridge.
       if (frame.Cypress) {
-        return frame.Cypress
+        // If the ending $ is not included in the template string, it breaks transpilation
```
Comment (chrisbreiding) [inline packages/runner/injection/multi-domain.js:19]:
```suggestion
        // If the ending $ is included in the template string, it breaks transpilation
```

It breaks it if it _is_ included, right? Does it still break if you escape the $ (e.g. `\$`)?

**Review by chrisbreiding — APPROVED** [review]

File: packages/driver/cypress/integration/e2e/multi-domain/multi_domain_validation_specs.ts [hunk 2]
```diff
@@ -198,6 +198,15 @@ describe('cy.origin', () => {
       .should('equal', 'Welcome TJohnson')
     })
 
+    it('finds the correct spec bridge even if a previous spec bridge host is a subset of the current host', () => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/integration/e2e/multi-domain/multi_domain_validation_specs.ts:201]: Given the PR title, I expect this test or a few tests to explicitly set chromeWebSecurity: false. Given we can't update that mid-test, it seems like a system-test should have been added to validate this??

File: packages/runner/injection/multi-domain.js (hunk 1 above)
Comment (mjhenkes) [reply packages/runner/injection/multi-domain.js:19]: My wording is confusing 🤪 .

This string breaks:
```
`(^|\\.)${ frame.location.host.replaceAll('.', '\\.') }$`
```

This string works: 
```
`(^|\\.)${ frame.location.host.replaceAll('.', '\\.') }` + '$'
```

File: packages/driver/cypress/integration/e2e/multi-domain/multi_domain_validation_specs.ts (hunk 2 above)
Comment (mjhenkes) [reply packages/driver/cypress/integration/e2e/multi-domain/multi_domain_validation_specs.ts:201]: The 'positive' case of the code is tested for each origin. We verify the port, protocol, and that the host ends with the host from the spec bridge. You are correct that the 'negative' is not tested. I made the call that testing this case  did not nessitate the cost of adding a system test (System tests seem expensive to me 🤷‍♂️ ). I can add a system test if the group disagrees with that reasoning.

Comment (mjhenkes) [reply packages/driver/cypress/integration/e2e/multi-domain/multi_domain_validation_specs.ts:201]: I've updated an existing system test for chromeWebSecurity to also test this case.

**Review by emilyrohrbough — APPROVED** [review]

**Review by AtofStryker — COMMENTED** [review]
was a little late 😆

**Pull request merged**